pkg/controllers/routing/aws.go

```diff
@@ -0,0 +1,69 @@
+package routing
+
+import (
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// disableSourceDestinationCheck disables src-dst check of all the VM's when cluster
+// is provisioned on AWS. EC2 by default drops any packets originating or destination
+// to a VM with IP other than that of VM's ip. This check needs to be disabled so that
+// cross node pod-to-pod traffic can be sent and recived by a VM.
+func (nrc *NetworkRoutingController) disableSourceDestinationCheck() {
```

Comment:
Is there a reason this has to iterate over all nodes and can't just check the current node?

Reply:
Agree, it should be improved. I have opened https://github.com/cloudnativelabs/kube-router/issues/407